src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -18,25 +23,68 @@
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
             + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
             + "Example: " + COMMAND_WORD + " alice bob charlie";
+    public static final String WARNING_INVALID_CONTEXT = "This command can only be used"
+            + "while viewing the list of activities/contacts.";
+    private final String searchTerm;
+    private final String[] keywords;
 
-    private final NameContainsKeywordsPredicate predicate;
-
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    public FindCommand(String[] keywords, String searchTerm) {
+        this.keywords = keywords;
+        this.searchTerm = searchTerm;
+        System.out.print("Keywords: ");
+        for (String a : keywords) {
+            System.out.print(a + ",");
+        }
+        System.out.println("");
+        System.out.println("searchTerm: " + searchTerm);
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
-        return new CommandResult(
-                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+        ContextType type = model.getContext().getType();
+        switch (type) {
+        case LIST_CONTACT:
+            model.updateFilteredPersonList(new NameContainsKeywordsPredicate(Arrays.asList(keywords)));
+            return new CommandResult(String.format(Messages.MESSAGE_FOUND_BY_KEYWORD,
```

Comment:
Odd that there's no empty line here but the corresponding `LIST_ACTIVITY` case has one.

Reply:
Fixed linting